lib/uniris/networking/networking.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Uniris.Networking do
+  @moduledoc """
+  Module defines networking configuration of the node.
+  """
+
+  alias __MODULE__.{Config, IPLookup}
+
+  # Public
+  
+  @doc """
+  Provides current host IP address.
+  """
+  @spec get_node_ip() :: {:ok, :inet.ip_address()} | {:error, :invalid_ip_provider | :not_recognizable_ip | :ip_discovery_error}
+  defdelegate get_node_ip, to: IPLookup
+
+  @doc """
+  Provides P2P port number.
+  """
+  @spec get_p2p_port() :: {:ok, pos_integer} | {:error, :invalid_port | :port_unassigned}
+  def get_p2p_port do
```

Comment:
I think the best should be to:
1. Get port from config if possible
2. Then try to publish it
3. If not, use random port and try to publish

Reply:
ok

Reply:
I'm not sure it will work in a cloud. Where there is no UPnP available and you need just a strict port. In such a case steps 2 and 3 will mess with the security group/ACL settings of the instance.

Reply:
We could use switch to identify the strategy to do given the ip provider.
If NAT, try the UPnP port publication (default to the one inside the conf)
if Static, just use the port
if IPFY, just use the port (maybe to test if the port is opened otherwise it's an error)

Those are just example of adaptative behaviors

Reply:
I propose the next algorithm:
  1. No port in config && UPnP or NAT PMP is available -> get random port from the pool.
  2. Port in config && UPnP or NAT PMP is available - try to publish port from config.
  3. Port in config && Unable to publish port from config && UPnP or NAT PMP is available - get random port from the pool.
  4. Port in config && UPnP or NAT PMP not available -> return port from config.
  5. No port in config && UPnP or NAT PMP not available -> find open port.
  6. No port in config && open port not found && UPnP or NAT PMP not available -> return error.

Reply:
Some of those points are good but I don't think the case when the port is missing in the conf will really happen. 

The default conf will rarely change in production. 

So we can assume than the port should be present otherwise it will crash (as a specification in the let's crash philosophy)

Reply:
We could also log the fact than the port is not available and request the user to open manually the port if not possible